src/azure-cli/azure/cli/command_modules/appconfig/_params.py

```diff
@@ -209,63 +209,68 @@ def load_arguments(self, _):
         c.argument('key', help='If no key specified, return all keys by default. Support star sign as filters, for instance abc* means keys with abc as prefix.')
         c.argument('label', help="If no label specified, list all labels. Support star sign as filters, for instance abc* means labels with abc as prefix. Use '\\0' for null label.")
 
+    with self.argument_context('appconfig feature') as c:
+        c.argument('key', validator=validate_feature_key, help='Key of the feature flag. Default key is the reserved prefix ".appconfig.featureflag/" + feature name.')
+
     with self.argument_context('appconfig feature show') as c:
-        c.argument('feature', help='Name of the feature flag to be retrieved')
+        c.argument('feature', help='Name of the feature flag to be retrieved. If the feature flag key is different from the default key, provide the `--key` argument instead.')
         c.argument('label', help="If no label specified, show entry with null label. Filtering is not supported.")
```

Comment:
We don't validate feature here?

Reply:
We never validated feature in any command except `set`. The reason was that we had some superficial restrictions on feature name that were not enforced by backend, so it was possible to have feature names without any character restrictions. 
However, for `--key` argument, we should always validate input because we want to make sure that it always starts with the reserved feature flag prefix.

Reply:
Got it. I guess previously feature name should at least be required. But with this change, we check the existance along with key.